xilem_web/src/concurrent/async_repeat.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{future::Future, marker::PhantomData, sync::Arc};
+
+use futures::channel::oneshot;
+use wasm_bindgen_futures::spawn_local;
+use xilem_core::{MessageResult, Mut, NoElement, View, ViewId, ViewMarker};
+
+use crate::{context::MessageThunk, DynMessage, Message, ViewCtx};
+
+pub fn async_repeat<M, F, H, State, Action, Fut>(
+    future_future: F,
+    on_event: H,
+) -> AsyncRepeat<F, H, M>
+where
+    F: Fn(AsyncRepeatProxy, oneshot::Receiver<()>) -> Fut + 'static,
```

Comment:
There might be a better API than passing `oneshot::Receiver<()>` as an abort handle, but for now at least it works :stuck_out_tongue_winking_eye:

Reply:
Oh right, it's not possible to cancel promises without something like this, unfortunate (though on the other hand also allows a more graceful shutdown)...
Yeah in that case that's probably ok. I wonder whether that should be a new type wrapper to make it more obvious, but I fear that needs quite a bit more code.

So this should probably documented then.

Reply:
> I wonder whether that should be a new type wrapper to make it more obvious, but I fear that needs quite a bit more code.

I made a simple wrapper that should help a bit :smile: